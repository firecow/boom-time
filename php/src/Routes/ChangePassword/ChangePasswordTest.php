<?
declare(strict_types=1);

namespace App\Routes\ChangePassword;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class ChangePasswordTest extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPAuthorization("Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTIyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6Ijc0NTA3NTFmMzA3YmM3NmQ5NjBmMmQ3OTJjNjZmNzIwIiwiaW1nIjoiaHR0cDovL2ZpcmVjb3cuZGsvaW1hZ2VzL2JnLnBuZyIsInVuYSI6IkZpcmVjb3cifQ.uG03s_iBmB2Zmd3ZBf-MAnlmyWPRSPsUT9FLoH7CSGk");
        $superGlobals->setHTTPPostValue("oldPassword", "ikeasucks");
        $superGlobals->setHTTPPostValue("newPassword", "newpasswordisawesomesauce");
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/changepassword/");
        return $routeResponse->prettyPrint();
    }
}
