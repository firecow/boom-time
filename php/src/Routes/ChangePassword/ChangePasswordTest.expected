HTTP/1.1 200 Ok
Content-Length: 16
Content-Type: application/json

{
    "success": true
}

- INSERT INTO `user_passwords` VALUES ('7450751f307bc76d960f2d792c66f720','$2y$10$Pys/vUso4uzu69Z.cxvgP.QoMNY.d1lx8RqUWEDqW3WbBE7JvNviW');
+ INSERT INTO `user_passwords` VALUES ('7450751f307bc76d960f2d792c66f720','2b5ba3ada83baba331dd86cbb5d78dc2');
